pac4j-oauth/src/main/java/org/pac4j/scribe/extractors/WeiboJsonExtractor.java

```diff
@@ -0,0 +1,36 @@
+package org.pac4j.scribe.extractors;
+
+import java.util.regex.Pattern;
+
+import org.pac4j.scribe.model.WeiboToken;
+
+import com.github.scribejava.core.extractors.OAuth2AccessTokenJsonExtractor;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+
+/**
+ * This class represents a specific JSON extractor for Weibo using OAuth protocol version 2. It
+ * could be part of the Scribe library.
+ *
+ * @author Zhang Zhenli
+ * @since 3.1.0
+ */
+public class WeiboJsonExtractor extends OAuth2AccessTokenJsonExtractor {
```

Comment:
I'm surprised not to see this class in the original Scribe Java library, but I guess this was necessary when doing your tests.

Reply:
Oh! I guess I get it: you wanted to extract the UID...

Reply:
Yes